static/js/thing.js

```diff
@@ -32,6 +32,7 @@ var Thing = function(description, format) {
   // Parse base URL of Thing
   if (description.href) {
     this.href = new URL(description.href, App.ORIGIN);
+    this.id = this.href.pathname.split('/').pop();
```

Comment:
Just a warning that this may not work once we allow adding things by URL... but we can cross that bridge when we come to it.